userguide/snippets/pathlengths.rst

```diff
@@ -34,11 +34,27 @@ can also reduce the length of these limits by one or more bytes.
    | Names               |                |                                                                        |
    +---------------------+----------------+------------------------------------------------------------------------+
    | Mounted Filesystem  | 88 bytes       | Mounted filesystem path length (*MNAMELEN*). Longer paths can prevent  |
-   | Paths               |                | a device from being mounted.                                           |
+   | Paths               |                | a device from being mounted.  See below for examples of path length    |
+   |                     |                | calculation.                                                           |
    +---------------------+----------------+------------------------------------------------------------------------+
    | Device Filesystem   | 63 bytes       | `devfs(8)                                                              |
    | Paths               |                | <https://www.freebsd.org/cgi/man.cgi?query=devfs&sektion=8>`__ device  |
    |                     |                | path lengths (*SPECNAMELEN*). Longer paths can prevent a device from   |
    |                     |                | being created.                                                         |
    +---------------------+----------------+------------------------------------------------------------------------+
 
+Care is needed with regard to ZFS snapshots. **If the mounted path length for a snapshot exceeds 88 characters, 
+then the snapshot and its data will be safe but inaccessible until its mounted path length is reduced**. 
+Typically this is rectified by renaming (shortening) the dataset, snapshot, or mountpoint, so that the path length is
```

Comment:
Again, this seems to imply that the mountpoint can be renamed without unmounting the snapshot.  Is that correct?

Reply:
See comment above, reworded to be clearer.

Reply:
"Path lengths that exceed 88 characters will cause data to be unavailable until the path to the device is shortened." 
The original text referred to snapshots but the table above seems to indicate that we are talking devices and the data on the device. 
"Correcting the path to be less than the 88 character limit will allow access to the data on the device."
The end of the sentence was split into the next comment. The text flipped back to the "88 byte" limit. We need to confirm if there is an 88 character, or byte limit and use the correct term.

Reply:
Yes. if a snapshot is unmountable, the symptom that the user will notice is not a "snapshot cannot be mounted" error, but that the data usually exposed by that snapshot is not being exposed as expected. In other words the error they perceive will be that their data is not accessible, and the reassurance they will want is that their data is intact anyhow. So focus on the implications for *data in the snapshot* is appropriate as it's clearer on these aspects. For that reason I feel it's important to lead with what the user most needs to know - **their data is safe but inaccessible** until the length issue is fixed. Skipping over it with a positive sounding "correcting the path will allow access to the data" doesn't explicitly confirm that snaps are intact, and for a user who suddenly can't get to the snapshots, that is one of the first things they will want to read explicit reassurance about. Once they know it's safe, they will next ask, how to recover it. "Safe but inaccessible" is the 1st thing anyone will want to read, in this case.